internal/app/pactproxy/response_modification_writer.go

```diff
@@ -7,36 +7,46 @@ import (
 )
 
 type ResponseModificationWriter struct {
-	res          http.ResponseWriter
-	interactions []*interaction
-	statusCode   int
+	res              http.ResponseWriter
+	interactions     []*interaction
+	originalResponse []byte
```

Comment:
Is this change included in #17 separately? We should merge that first if so

Reply:
Yes exactly. We merged the branch `liesk-naveen-fix-response-modifications` to this one so we can test it in swift-gw. The plan is to first merge https://github.com/form3tech-oss/pact-proxy/pull/17 and then proceed with this PR (rebase etc.) :)